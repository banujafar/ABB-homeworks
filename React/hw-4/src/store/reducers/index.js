import { combineReducers } from "redux";
import { modalReducer } from "./modalReducer";
import { productReducer } from "./productReducer";
import { cartReducer } from "./cartReducer";
import { favoritesReducer } from "./favoritesReducer";

export const rootReducer=combineReducers({
    modal:modalReducer,
    products:productReducer,
    cart:cartReducer,
    favorites:favoritesReducer
})