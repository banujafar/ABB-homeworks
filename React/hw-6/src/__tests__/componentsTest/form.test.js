import FormCart from "../../components/Form/Form";
import { Provider } from "react-redux";
import { store } from "../../store";
import { render } from "@testing-library/react";

test("should  matches DOM snapshot correctly", () => {
  const { container } = render(
    <Provider store={store}>
      <FormCart />
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
