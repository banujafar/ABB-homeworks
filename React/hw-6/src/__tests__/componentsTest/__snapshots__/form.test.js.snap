// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  matches DOM snapshot correctly 1`] = `
<div>
  <form
    class="form"
  >
    <label
      for="firstName"
    >
      First Name
    </label>
    <input
      class=""
      id="firstName"
      name="firstName"
      placeholder="Enter your first name"
      type="text"
      value=""
    />
    <label
      for="lastName"
    >
      Last Name
    </label>
    <input
      class=""
      id="lastName"
      name="lastName"
      placeholder="Enter your last name"
      type="text"
      value=""
    />
    <label
      for="age"
    >
      Age
    </label>
    <input
      class=""
      id="age"
      name="age"
      placeholder="Enter your age"
      type="number"
      value=""
    />
    <label
      for="address"
    >
      Delivery Address
    </label>
    <input
      class=""
      id="address"
      name="address"
      placeholder="Enter your delivery address"
      type="text"
      value=""
    />
    <label
      for="phone"
    >
      Phone
    </label>
    <input
      class=""
      id="phone"
      name="phone"
      placeholder="Enter your phone number"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Checkout
    </button>
  </form>
</div>
`;
